tests/zfs-tests/cmd/randfree_file/randfree_file.c

```diff
@@ -98,11 +98,13 @@ main(int argc, char *argv[])
 
 	free(buf);
 
+#if defined(FALLOC_FL_PUNCH_HOLE) && defined(FALLOC_FL_KEEP_SIZE)
 	if (fallocate(fd, FALLOC_FL_PUNCH_HOLE | FALLOC_FL_KEEP_SIZE,
 	    start_off, off_len) < 0) {
 		perror("fallocate");
 		return (1);
 	}
+#endif /* defined(FALLOC_FL_PUNCH_HOLE) && defined(FALLOC_FL_KEEP_SIZE) */
```

Comment:
This does fix the build issue but it also silently breaks this utility for older kernels.  Let's add an `#else` case to the conditional to print an error message along the lines of "FALLOC_FL_PUNCH_HOLE unsupported" and `return (1);` to indicate an error.